test/public/envs/overview.test.js

```diff
@@ -130,24 +129,20 @@ module.exports = () => {
     });
 
     it('can set how many environments are available per page', async () => {
-        await waitForTimeout(300);
         // Expect the amount selector to currently be set to 10 (because of the defined page height)
         const amountSelectorId = '#amountSelector';
         const amountSelectorButton = await page.waitForSelector(`${amountSelectorId} button`);
         const amountSelectorButtonText = await page.evaluate((element) => element.innerText, amountSelectorButton);
-        await waitForTimeout(300);
         expect(amountSelectorButtonText.trim().endsWith('10')).to.be.true;
 
         // Expect the dropdown options to be visible when it is selected
         await amountSelectorButton.evaluate((button) => button.click());
-        await waitForTimeout(100);
-        const amountSelectorDropdown = await page.$(`${amountSelectorId} .dropup-menu`);
-        expect(Boolean(amountSelectorDropdown)).to.be.true;
+        await page.waitForSelector(`${amountSelectorId} .dropup-menu`);
 
         // Expect the amount of visible environments to reduce when the first option (5) is selected
         const menuItem = await page.$(`${amountSelectorId} .dropup-menu .menu-item`);
         await menuItem.evaluate((button) => button.click());
-        await waitForTimeout(100);
+        await waitForTableLength(page, 5);
 
         const tableRows = await page.$$('table tr');
         expect(tableRows.length - 1).to.equal(5);
```

Comment:
Doing so, you can remove lines 147 and 148

Reply:
Done